utPlayer.py

```diff
@@ -31,42 +31,64 @@
 
 from framework.core.logModule import logModule
 from framework.plugins.ut_raft.interactiveShell import InteractiveShell
+from framework.plugins.ut_raft.configRead import ConfigRead
+
 class utPlayer():
     """
     UT Player class
     """
-    def __init__(self, session:object, playerTool:dict={'tool': 'gstreamer', 'prerequisites':[]}, log:logModule=None):
+    playerConfig = os.path.join(dir_path, "uPlayerConfig.yml")
+
+    def __init__(self, session:object, platform:str, player:str = "gstreamer", log:logModule=None):
         """
         Initializes player class.
 
         Args:
             session (class): The session object to communicate with the device
+            platform (str): Platfrom name
+            player (str, optional) : Player to use. Defaults to gstreamer
             log (class, optional): Parent log class. Defaults to None.
         """
         self.log = log
         if log is None:
             self.log = logModule(self.__class__.__name__)
             self.log.setLevel( self.log.INFO )
         self.session = session
-        self.playbackTool = playerTool["tool"]
-        for cmd in playerTool["prerequisites"]:
-            self.session.write(cmd)
+        self.playerProfile = ConfigRead(self.playerConfig, platform)
+        self.player = self.playerProfile.get(player)
+
+        if self.player.get("prerequisites"):
+            for cmd in self.player.get("prerequisites"):
+                self.session.write(cmd)
 
-    def play(self, streamFile:str):
+    def play(self, streamFile:str, mixer_input:str = "primary"):
         """
         Starts the playback of a stream
         Once started the stream is assumed to be blocking on the device.
         Unless an error occurs, it will remain running
 
         Args:
             stream (str): Stream path.
+            mixer_input (str, optional): Mixer input to which audio output should be connected. Defaults to "primary"
         """
         #TODO: Upgrade if required or a new function to playback from a URL
         # Example usage for gst-launch `gst-launch-1.0 filesrc location=/home/yourusername/myvideo.mp4 ! decodebin ! autovideosink`
         # Example usage for gst-play `gst-play-1.0 <file_path>`
-        if (self.playbackTool == "gstreamer"):
-            cmd = "gst-play-1.0" + " " + streamFile
-            self.session.write(cmd)
+
+        if mixer_input == "primary":
+            cmd = self.player.get("play_command")
+            if not cmd:
+                self.log.error("Player command not found")
+                return
+            cmd += " " + streamFile
+        elif mixer_input == "secondary":
+            cmd = self.player.get("secondary_play_command")
+            if not cmd:
+                self.log.error("Secondary Player command not found")
+                return
+            cmd += " " + streamFile
```

Comment:
Once you move this decision logic to it's own function this just becomes.

```python
cmd +=  self.mixerInputConfiguration + streamFile
```

mixerConfiguration - set by the function setMixerInput() - function